pkg/pgmodel/metrics/database/metrics.go

```diff
@@ -554,6 +554,44 @@ var metrics = []metricQueryWrap{
 		),
 		query: `select count(*)::bigint from _prom_catalog.metric`,
 	},
+	{
+		metrics: gauges(
+			prometheus.GaugeOpts{
+				Namespace: util.PromNamespace,
+				Subsystem: "sql_database",
+				Name:      "shared_buffers_size",
+				Help:      "Size of shared_buffers in bytes",
+			},
+		),
+		query: `SELECT (setting::BIGINT*pg_size_bytes(unit))::BIGINT FROM pg_settings WHERE name = 'shared_buffers'`,
+	},
+	{
+		metrics: gauges(
+			prometheus.GaugeOpts{
+				Namespace: util.PromNamespace,
+				Subsystem: "sql_database",
+				Name:      "open_chunks_total_table_size",
+				Help:      "Total table size of currently open chunks in bytes",
+			},
+			prometheus.GaugeOpts{
+				Namespace: util.PromNamespace,
+				Subsystem: "sql_database",
+				Name:      "open_chunks_total_index_size",
+				Help:      "Total indexes size of currently open chunks in bytes",
+			},
+		),
+		query: `SELECT 
```

Comment:
Looking at the explain of the query, it doesn't look great.
I think the following query is much more performant 
```
explain (verbose, ANALYZE)
SELECT
  coalesce(sum(chunk_total_size)::BIGINT,0) as total_table_size,
  coalesce(sum(chunk_index_size)::BIGINT, 0) as total_index_size
FROM
(
select 
  pg_indexes_size(format('%I.%I', c.schema_name, c.table_name)) chunk_index_size,
  pg_total_relation_size(format('%I.%I', c.schema_name, c.table_name)) chunk_total_size
from _timescaledb_catalog.dimension_slice ds 
inner join _timescaledb_catalog.chunk_constraint cc on (cc.dimension_slice_id = ds.id)
inner join _timescaledb_catalog.chunk c on (c.id = cc.chunk_id) 
where  range_end > _timescaledb_internal.time_to_internal(now()) 
and range_start < _timescaledb_internal.time_to_internal(now())
) as x;
```

Reply:
But I think the above is not very accurate since `range_end > _timescaledb_internal.time_to_internal(now()) ` may not match all tables if we fall even a bit behind. I suggest
```
explain (verbose, ANALYZE)
SELECT
  coalesce(sum(chunk_total_size)::BIGINT,0) as total_table_size,
  coalesce(sum(chunk_index_size)::BIGINT, 0) as total_index_size
FROM
(
select DISTINCT ON (hypertable_id)
  pg_indexes_size(format('%I.%I', c.schema_name, c.table_name)) chunk_index_size,
  pg_total_relation_size(format('%I.%I', c.schema_name, c.table_name)) chunk_total_size
from _timescaledb_catalog.dimension_slice ds 
inner join _timescaledb_catalog.chunk_constraint cc on (cc.dimension_slice_id = ds.id)
inner join _timescaledb_catalog.chunk c on (c.id = cc.chunk_id) 
where  range_end > _timescaledb_internal.time_to_internal(now()- interval '1 hour') 
and range_start < _timescaledb_internal.time_to_internal(now())
order by hypertable_id, range_end desc
) as x;
```
which corrects for that

Reply:
I actually had similar query mostly borrowed from the view `timescaledb_information.chunk`. It is faster but like your query above it uses TSDB internal catalogs and functions so I've decided to go with the `public` one as former can break on TSDB changes. I fine with switching just wanted to lay out the tradeoffs since this query runs every 3 minutes. I'll also check with Valeriy.

Reply:
Btw on 2M chunks query takes around 150-200ms.

Reply:
I've added 5 minutes if we fall a bit behind. IIRC from benchmarks we rarely seen more then few minutes of lag.